server/api/api.go

```diff
@@ -549,6 +550,83 @@ func (a *API) handleDeleteBlock(w http.ResponseWriter, r *http.Request) {
 	auditRec.Success()
 }
 
+func (a *API) handlePatchBlock(w http.ResponseWriter, r *http.Request) {
+	// swagger:operation PATCH /api/v1/workspaces/{workspaceID}/blocks/{blockID} patchBlock
+	//
+	// Partially updates a block
+	//
+	// ---
+	// produces:
+	// - application/json
+	// parameters:
+	// - name: workspaceID
+	//   in: path
+	//   description: Workspace ID
+	//   required: true
+	//   type: string
+	// - name: blockID
+	//   in: path
+	//   description: ID of block to patch
+	//   required: true
+	//   type: string
+	// - name: Body
+	//   in: body
+	//   description: block patch to apply
+	//   required: true
+	//   schema:
+	//     "$ref": "#/definitions/BlockPatch"
+	// security:
+	// - BearerAuth: []
+	// responses:
+	//   '200':
+	//     description: success
+	//   default:
+	//     description: internal error
+	//     schema:
+	//       "$ref": "#/definitions/ErrorResponse"
+
+	ctx := r.Context()
+	session := ctx.Value(sessionContextKey).(*model.Session)
+	userID := session.UserID
```

Comment:
nit and non-specific to this PR: I think we could abstract this in a couple of helpers like `getSessionFromRequest(r)` and `getUserIDFromRequest(r)`, as I'm seeing this three lines repeated all over `api`

Reply:
I like the explicit better than implicit. Anyway, we can decide to move in that direction but I would prefer to do the migration all at once in another PR.

Reply:
Agree on postponing this to another PR, but I don't think we're hiding any complexity here but simplifying something that we're doing almost everywhere. We already have helpers that go in that direction, like [`GetContextConn`](https://github.com/mattermost/focalboard/blob/main/server/api/context.go#L22), so I'm thinking in something close to that